### PR TITLE
18 mapping device information obx 18 equipment instance identifier

### PR DESCRIPTION
**Pull request by d-gregorczyk**

### CODE REVIEW

**Review by riechkathrin — APPROVED** [review]

**Comment by d-gregorczyk** [comment]
Please thoroughly review, esp. the changes/insertions w.r.t to the extension (XML Schema plus requirements).

Apart from the scope of the ticket, I also provided

- Offline font files, which might have been an issue formerly since on each visit of the supplement there have been requests to Google servers
- Syntax highlighting
- A XML Schema namespace-prefix-mapping table (@ToddCooper please verify where to move the table to. It is required by Volume 2 and 3)

**Comment by ToddCooper** [comment]
> Please thoroughly review, esp. the changes/insertions w.r.t to the extension (XML Schema plus requirements).
> 
> * A XML Schema namespace-prefix-mapping table (@ToddCooper please verify where to move the table to. It is required by Volume 2 and 3)

Reviewing the ITI TF, here is what I came up with ...

**IHE "Namespace"** 

- In general this always points back to OID namespaces (which we also have to do) and not URI / WS namespaces.  See [TF-0 IHE Domain Namespaces](https://profiles.ihe.net/GeneralIntro/ch-C.html#C.1)

[**ITI TF-2 Appendix V:  Web Services for IHE Transactions**](https://profiles.ihe.net/ITI/TF/Volume2/ch-V.html)

- This provides general profiling guidance but nothing around XML .
- [V.2.4 XML Namespaces ](https://profiles.ihe.net/ITI/TF/Volume2/ch-V.html#V.2.4)provides a table for specific prefixes 
- [V.3.2.1.1 Naming conventions and namespaces](https://profiles.ihe.net/ITI/TF/Volume2/ch-V.html#V.3.2.1.1) also provides conventions for ITI's use

[**ITI TF-3 6.1 IHE ITI Namespaces**](https://profiles.ihe.net/ITI/TF/Volume3/ch-6.1.html#6.1)

- Simply points to the IHE wiki domain namespaces table.
- We would point to something similar but for our IHE DEV namespace extensions OID table 

To summarize, TF-2 is the only clear place where there is an analogous support for XML namespaces used in ITI's profiles and we should do the same.  

**QUESTION**:  Is whether we should include this as part of TF-2 Appendix A or if we see it as needing its own Appendix as ITI did with Appendix V.  

- Then the question is whether this should be a new appendix or folded into A.
- Since it is across both TF-2 Appendix A and Appendix B, we should probably have it in a separate, common place.  
- Wherever that is should reference the ITI TF-2 Appendix V for conventions and then use the same table format as V.2.4 above.  

**Question**:  Do we have a similar issue in TF-3 BICEPS content spec's?  If so, then do we need one also in TF-3 too?  And if so, are those different from those in TF-2?

**Comment by d-gregorczyk** [comment]
As said, the namespace mapping is needed by both TF-2 and TF-3. The content modules depend on XML Schema, since the BICEPS domain model is based on XML Schema, which utilizes XML namespaces.

I just want to point out that by putting the mapping table to TF-2, we are changing the dependency hierarchy - TF-3 now depends on TF-2.

Alternatively, I can create a adoc file from the namespace mapping table and include the file in TF-2 and TF-3. Then however we will have two places with the same information. I kind of dislike that.

File: asciidoc/volume3/biceps-extension-provisions/tf3-ch-8.3.2.9-extension-equipment-identifier.adoc [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+.R0017
+[sdpi_requirement#r0017,sdpi_req_level=shall]
+****
+A <<actor_somds_provider>> shall add the `sdpi:EquipmentIdentifier` extension to each pm:MdsDescriptor in its <<acronym_mdib>>.
+****
+
+.R0018
+[sdpi_requirement#r0018,sdpi_req_level=should]
+****
+A <<actor_somds_provider>> should add the `sdpi:EquipmentIdentifier` extension to each pm:VmdDescriptor in its <<acronym_mdib>>.
```
Comment (MartinKasp) [inline asciidoc/volume3/biceps-extension-provisions/tf3-ch-8.3.2.9-extension-equipment-identifier.adoc:53]: Analog to my comment on R0017, this is quite strict for a VMD. "May" might be useful or a condition, e.g., to removable subsystems represented by an VMD?

**Review by MartinKasp — COMMENTED** [review]
Discussion needed on binding strength of expansion sdpi:EquipmentIdentifier. See particular comments.

File: asciidoc/volume3/biceps-extension-provisions/tf3-ch-8.3.2.9-extension-equipment-identifier.adoc (hunk 1 above)
Comment (d-gregorczyk) [reply asciidoc/volume3/biceps-extension-provisions/tf3-ch-8.3.2.9-extension-equipment-identifier.adoc:53]: I'm going to add a condition as you proposed.

File: asciidoc/volume3/biceps-extension-provisions/tf3-ch-8.3.2.9-extension-equipment-identifier.adoc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+.R0016
+[sdpi_requirement#r0016,sdpi_req_level=shall]
+****
+To test if two equipment identifiers are equal, a <<actor_somds_participant>> shall perform case-sensitive string comparison.
+****
+
+.R0017
+[sdpi_requirement#r0017,sdpi_req_level=shall]
+****
+A <<actor_somds_provider>> shall add the `sdpi:EquipmentIdentifier` extension to each pm:MdsDescriptor in its <<acronym_mdib>>.
```
[the remark this answers is not in the document]
Comment (d-gregorczyk) [reply asciidoc/volume3/biceps-extension-provisions/tf3-ch-8.3.2.9-extension-equipment-identifier.adoc:47]: It is mandatory in order to allow for a consumer to rely on the ID to be present in the MDIB. If the UDI sufficiently covers identity, the provider can generate a UUID from the UDI; or can combine manufacturer namespace plus serialnumber name to a UUID. I'm going add a note for clarification.
